compiler/frontend/src/org/jetbrains/kotlin/resolve/diagnostics/KotlinSuppressCache.kt

```diff
@@ -35,6 +35,7 @@ import org.jetbrains.kotlin.util.ExtensionProvider
 
 interface DiagnosticSuppressor {
     fun isSuppressed(diagnostic: Diagnostic): Boolean
+    fun isSuppressed(diagnostic: Diagnostic, bindingContext: BindingContext?): Boolean = isSuppressed(diagnostic)
```

Comment:
Why `bindingContext` is nullable here?
Note that we have `BindingContext.EMPTY` that can be used when there is no binding context

Reply:
In my mind, there was/is a difference between a bindingContext that doesn't have the slices resolved, and being in a situation where a bindingContext is not supported.  In our case, if the bindingContext is null, we follow a different best-effort code path.

I actually don't feel strongly here.  Happy to change if BindingContext.EMPTY is more desirable.  For our best-effort codepath, we can just check if the bindingContext is equal to BindingContext.EMPTY.  Would you like me to change it to use BindingContext.EMPTY?

Longer term, I think a better solution might be to get rid of the `fun isSuppressed(diagnostic: Diagnostic)` in favor of `fun isSuppressed(diagnostic: Diagnostic, bindingContext: BindingContext)` and remove nullability.  Such that you can't have a DiagnosticSuppressor without being able to get a context for the given diagnostic.  That was a potentially bigger / moredisruptive change, which is why we opted for the less disruptive variant.  Thoughts?

Reply:
The current implementation is totally fine, I just wanted to figure out why it's done this way. 
In addition, we have freedom here to change the method's signature (or add a new method) if necessary, so I think we can now take the current approach